Mathlib/Analysis/Meromorphic/Divisor.lean

```diff
@@ -66,6 +68,47 @@ Simplifier lemma: on `U`, the divisor of a function `f` that is meromorphic on `
 lemma divisor_apply {f : 𝕜 → E} (hf : MeromorphicOn f U) (hz : z ∈ U) :
     divisor f U z = (hf z hz).order.untop₀ := by simp_all [MeromorphicOn.divisor_def, hz]
 
+/-!
+## Congruence Lemmas
+-/
+
+/--
+If `f₁` is meromorphic on `U`, if `f₂` agrees with `f₁` on a codiscrete subset of `U` and outside of
+`U`, then `f₁` and `f₂` induce the same divisors on `U`.
+-/
+theorem divisor_congr_codiscreteWithin_of_eqOn_compl {f₁ f₂ : 𝕜 → E} (hf₁ : MeromorphicOn f₁ U)
+    (h₁ : f₁ =ᶠ[Filter.codiscreteWithin U] f₂) (h₂ : Set.EqOn f₁ f₂ Uᶜ) :
+    divisor f₁ U = divisor f₂ U := by
+  ext x
+  by_cases hx : x ∈ U <;> simp [hf₁, hf₁.congr_codiscreteWithin_of_eqOn_compl h₁ h₂, hx]
+  · congr 1
+    apply (hf₁ x hx).order_congr
+    simp_rw [EventuallyEq, Filter.Eventually, mem_codiscreteWithin,
+      disjoint_principal_right] at h₁
+    filter_upwards [h₁ x hx] with a ha
+    simp at ha
+    tauto
+
+/--
+If `f₁` is meromorphic on an open set `U`, if `f₂` agrees with `f₁` on a codiscrete subset of `U`,
+then `f₁` and `f₂` induce the same divisors on`U`.
+-/
+theorem divisor_congr_codiscreteWithin {f₁ f₂ : 𝕜 → E} (hf₁ : MeromorphicOn f₁ U)
+    (h₁ : f₁ =ᶠ[Filter.codiscreteWithin U] f₂) (h₂ : IsOpen U) :
+    divisor f₁ U = divisor f₂ U := by
+  ext x
+  by_cases hx : x ∈ U <;> simp [hf₁, hf₁.congr_codiscreteWithin h₁ h₂, hx]
```

Comment:
ditto

Reply:
Fixed. To avoid overly long lines, I formulated the two invocations of `simp` separately.